### PR TITLE
Fix golint failures of e2e/framework/util.go - part2

### PR DESCRIPTION
**Pull request by atoato88**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This is a part of a series for fixing golint failures for util.go.
- fixes golint failures from line 1395 to line 2353 at original util.go

This fixes golint failures of the following file:
- test/e2e/framework/util.go

This changes following files because of change function name
in above file.
- test/e2e/apps/rc.go
- test/e2e/apps/replica_set.go

**Special notes for your reviewer**:

The others of this series are following:
- part1: #76348

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

Ref: #68026

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -2098,6 +2109,7 @@ func ServiceResponding(c clientset.Interface, ns, name string) error {
 	})
 }
 
+// RestclientConfig returns a config holds the information needed to build connect to kubernetes clusters.
```
Comment (SataQiu) [inline test/e2e/framework/util.go]:
```suggestion
// RestclientConfig returns a config holds the information needed to build connection to kubernetes clusters.
```

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -1515,7 +1516,7 @@ func podRunning(c clientset.Interface, podName, namespace string) wait.Condition
 	}
 }
 
-// WaitTimeoutForPodEvent waits for an event to occur for a pod
+// WaitTimeoutForPodEvent waits for an event to occur for a pod.
```
[the remark this answers is not in the document]
Comment (atoato88) [reply test/e2e/framework/util.go]: Thank you for comment.
I've fixed this.

File: test/e2e/framework/util.go [hunk 3]
```diff
@@ -1856,13 +1859,14 @@ type podProxyResponseChecker struct {
 	pods           *v1.PodList
 }
 
-func PodProxyResponseChecker(c clientset.Interface, ns string, label labels.Selector, controllerName string, respondName bool, pods *v1.PodList) podProxyResponseChecker {
-	return podProxyResponseChecker{c, ns, label, controllerName, respondName, pods}
+// GetPodProxyResponseChecker returns a context for checking a pods responses.
```
[the remark this answers is not in the document]
Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go (hunk 1 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go [hunk 4]
```diff
@@ -1856,13 +1859,14 @@ type podProxyResponseChecker struct {
 	pods           *v1.PodList
 }
 
-func PodProxyResponseChecker(c clientset.Interface, ns string, label labels.Selector, controllerName string, respondName bool, pods *v1.PodList) podProxyResponseChecker {
-	return podProxyResponseChecker{c, ns, label, controllerName, respondName, pods}
+// GetPodProxyResponseChecker returns a context for checking pods responses.
+func GetPodProxyResponseChecker(c clientset.Interface, ns string, label labels.Selector, controllerName string, respondName bool, pods *v1.PodList) PodProxyResponseChecker {
```
Comment (andrewsykim) [inline test/e2e/framework/util.go]: Is prepending this method with `Get` something that golint enforces?

Comment (andrewsykim) [reply test/e2e/framework/util.go]: Looks like this is a constructor method, what are your thoughts on `NewPodProxyResponseChecker` instead?

File: test/e2e/framework/util.go [hunk 5]
```diff
@@ -2251,40 +2267,46 @@ func KubectlCmd(args ...string) *exec.Cmd {
...
-type kubectlBuilder struct {
+type KubectlBuilder struct {
 	cmd     *exec.Cmd
 	timeout <-chan time.Time
 }
 
-func NewKubectlCommand(args ...string) *kubectlBuilder {
-	b := new(kubectlBuilder)
+// GetNewKubectlCommand returns a KubectlBuilder for running kubectl.
+func GetNewKubectlCommand(args ...string) *KubectlBuilder {
```
Comment (andrewsykim) [inline test/e2e/framework/util.go]: I liked `NewKubectlCommand` actually, was golint complaining about this?

File: test/e2e/framework/util.go (hunk 4 above)
Comment (oomichi) [reply test/e2e/framework/util.go]: This renaming seems for avoiding duplicated names between struct name (PodProxyResponseChecker which is returned with this method) and this method name.
`NewPodProxyResponseChecker` is also a good option for us I feel.

Comment (atoato88) [reply test/e2e/framework/util.go]: done.

File: test/e2e/framework/util.go (hunk 5 above)
Comment (atoato88) [reply test/e2e/framework/util.go]: Thank you for comment.
I've fixed this.

**Comment by atoato88** [comment]
I rebased with current master(commit: b359b6bfe7e49ae7bbb9d21552d2a06bc55f8bc2).

**Review by oomichi — COMMENTED** [review]
Thanks for updating

/lgtm

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Pull request merged**